### PR TITLE
WebSockets Next: support endpoints with empty path

### LINKED ISSUE #42808: quarkus-websockets-next client adds / to empty path
### Describe the bug

Using @WebSocketClient(path = "") with an empty path is not possible.
io.quarkus.websockets.next.runtime.WebSocketConnectorBase line 128:
`return path.startsWith("/") ? sb.toString() : "/" + sb.toString();`
This adds a / even if the path is empty

### Expected behavior

Add no / for an empty path

### Actual behavior

Adds a / to an empty path

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Linux endeavouros 6.10.6-arch1-1

### Output of `java -version`

openjdk version "21.0.4"

### Quarkus version or git rev

Version 3.13.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.8

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- fixes #42808

### CODE REVIEW

**Review by cescoffier — APPROVED** [review]

File: extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java [hunk 1]
```diff
@@ -123,6 +123,9 @@ Set<String> getPathParamNames(String path) {
     }
 
     String replacePathParameters(String path) {
+        if (path.isEmpty()) {
+            return "";
```
Comment (gsmet) [inline extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java:127]: I was going to add a comment to the processor but it's not really called often so I refrained to do it but in this case, I would simply return `path`.

**Review by gsmet — APPROVED** [review]
I added a small comment. Feel free to ignore and merge if you don't see value to it.

File: extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java (hunk 1 above)
Comment (mkouba) [reply extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java:127]: Well, `return "";` is compiled into bytecode that loads an entry from the constant pool. So the only difference is one more entry in the constant pool of `WebSocketConnectorBase` :shrug:.

Having said that, I don't know why I used this form here and I have no preference. So let's use `return path;` instead! ;-)

Comment (gsmet) [reply extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java]: Yeah I perfectly agree the cost is very small but small costs pile up and when there is absolutely no reason to pay this cost, I think we should avoid it.

Certainly not worth blocking the PR thus why I approved anyway and let you decide what to do :).

(I wouldn't have asked if it made the code worse or more complex)

**Review by gsmet — APPROVED** [review]

File: extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java (hunk 1 above)
Comment (mkouba) [reply extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectorBase.java]: > but small costs pile up 

You would need a loooot of `return "";` statements to see a very tiny cost but I get your point and the PR is updated now ;-).

> Certainly not worth blocking the PR thus why I approved anyway and let you decide what to do :).
> (I wouldn't have asked if it made the code worse or more complex)

:100:

**Pull request merged**